drivers/CoreBattery/source/CoreBattery.cpp

```diff
@@ -4,18 +4,22 @@
 
 #include "CoreBattery.h"
 
+#include "platform/Callback.h"
+
 #include "MathUtils.h"
 
 namespace leka {
 
-void CoreBattery::onChargeDidStart(mbed::Callback<void()> const &callback)
+void CoreBattery::onChargeDidStart(std::function<void()> const &callback)
 {
-	_charge_status_input.rise(callback);
+	_on_charge_did_start = callback;
+	_charge_status_input.rise(mbed::Callback<void()> {[this] { _on_charge_did_start(); }});
```

Comment:
is it not possible to do just that? Do we really need to go from std::function to lambda to mbed::callback?

```suggestion
	_charge_status_input.rise([this] { _on_charge_did_start(); });
```